### DISCUSSION TITLE
Bad Tag Handling in Obsidian

### DISCUSSION THREAD

**Discussion by popebabylon** [akosbalasko/yarle#313] [Q&A]
Hello,

I recently installed the macOS binary and tried a migration of a ~1400 notes enex file.  It mostly seemed to work, except that the tags did not port over to Obsidian at all.  The tags were in each file, but for some reason the tag section was bounded by ``` syntax which seemed to kill their treatment as tags.  Also, some random # mentions in other notes (which I think were just Twitter #'s saved in a web clip) were treated as tags by Obsidian.

I had selected:
Output format: ObsidianMD
Convert Webclips: true
Add # to tag: true

Is there something obvious I'm missing?  Or a best practice I should follow?

Thank you!

**Comment by akosbalasko** [comment]
hi @popebabylon , could you pls share the template and all of the settings you used?

**Comment by popebabylon** [comment]
Hello @akosbalasko -- I'm pasting the _original_ config I used at the end here (which was the default template recommended in the UI), but I have (for now) resolved the issue (with a bit of manual labor).

I was running in to a few problems.

1. Evernote allows you to use \`\`\` to fence codeblocks.  Markdown would prefer codeblocks identified by four leading spaces.  There was some confusion when these notes were imported because either:
    - The use of \`\`\` in the notes was sloppy, resulting in the Tag block (which was bounded by \`\`\`) sometimes being ignored (but sometimes not, weirdly)
    - Since some code uses #'s (like a #include in a simple C program), and the codeblock wasn't recognized, that was picked up as a tag
2. Some webclipped articles (usually about code) were using single \` to define codeblocks
    - Obsidian seemed to recognize that they were special (they _looked_ like codeblocks) but was still sometimes picking up #'s in them as Tags 
3. This isn't a yarle problem, but markdown/obsidian were being pretty aggressive at picking up other #'s as tags such as hex codes or twitter tags that I didn't want
    - If the # is in a URL or a clear codeblock it knows to ignore it, but anything ambiguous gets treated as a tag

I ended up changing the template to not use \`\`\` for the Tag block which helped.  Then flipped back and forth between Obsidian, BBEdit (for grep find/replace patterns), and looking at my existing Evernote to clean up bad codeblocks and all the other "incorrect" tags.  Not sure if there's an easier way.

So I guess my suggestion would be to not use \`\`\` in the default template?  And maybe see if you can recognize Evernote codeblocks and convert them more cleanly to markdown?  That might be tricky.

However this tool was still really helpful, thank you! 

`Current config is: {
    "enexSources": [
        "/Users/foo/Documents/notes.enex"
    ],
    "outputDir": "/Users/foo/Documents/notes",
    "keepOriginalHtml": false,
    "isMetadataNeeded": true,
    "isNotebookNameNeeded": false,
    "isZettelkastenNeeded": false,
    "plainTextNotesOnly": false,
    "skipWebClips": false,
    "useHashTags": true,
    "nestedTags": {
        "separatorInEN": "_",
        "replaceSeparatorWith": "/",
        "replaceSpaceWith": "-"
    },
    "outputFormat": "ObsidianMD",
    "urlEncodeFileNamesAndLinks": false,
    "pathSeparator": "/",
    "resourcesDir": "_resources",
    "turndownOptions": {
        "headingStyle": "atx"
    },
    "skipLocation": true,
    "skipCreationTime": false,
    "skipUpdateTime": false,
    "skipSourceUrl": false,
    "skipTags": false,
    "skipEnexFileNameFromOutputPath": true,
    "keepMDCharactersOfENNotes": false,
    "monospaceIsCodeBlock": true,
    "currentTemplate": "This is my template \n\n{title-block}# {title}{end-title-block}\n\n---\n\n\n\n{content-block}{content}{end-content-block}\n\n---\n\n{created-at-block}_Created at {created-at}._{end-created-at-block}\n{updated-at-block}_Last updated at {updated-at}._{end-updated-at-block}\n{source-url-block}_Source URL: []({source-url})._{end-source-url-block}\n\n\n{tags-block}\nTagged: \n```\n{tags}\n```\n{end-tags-block}\n",
    "dateFormat": "YYYY-MM-DD",
    "keepImageSize": "ObsidianMD",
    "addExtensionToInternalLinks": true
}`
